### PR TITLE
IAPI readme + py examples

### PR DESCRIPTION
**Pull request by johntmyers**

### CODE REVIEW

File: docs/inference_api_beta/README.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+Within this directory there are examples with Javascript and Python that provide lighweight wrappers around the API.
+
+# General Operation
+
+Gretel Tabular LLM utilizes lightweight streams to generate and retrieve tabular data. Compared to other LLM streaming mechanisms, such as Server-Sent Events (SSE), Gretel does not stream back individual tokens. The Tabular LLM infrastructure automatically utilizes multiple LLMs and Agents to generate tokens and re-assemble them into tabular data. Client-side applications can then poll specific REST API endpoints to retrieve table rows as they are generated and assembled. By using a lightweight polling mechanism, Tabular LLM can be more flexibly integrated into apps that may not have streaming capabilities.
+
+There are two main steps to utilize the Tabular LLM APIs:
+
+1) First, a request is made that contains the prompt and LLM parameters. This request will return immediately with a *inference stream ID*. This ID represents a lightweight stream that can be itereated next. Behind the scenes, this request will be processed and tabular records will be generated and written to the stream.
```
Comment (matthewgrossman) [inline docs/inference_api_beta/README.md:13]:
```suggestion
1) First, a request is made that contains the prompt and LLM parameters. This request will return immediately with a *inference stream ID*. This ID represents a lightweight stream that can be iterated through. Behind the scenes, this request will be processed and tabular records will be generated and written to the stream.
```

File: docs/inference_api_beta/README.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+In order to use this API you will need a [Gretel API key](https://console.gretel.ai/users/me/key). This API key will need to be included in all API call [HTTP headers](https://api.docs.gretel.ai/). If you are using the Javascript examples, you can set the API key as an enviroment variable. If you are using the Python examples, we highly suggest you install the [Gretel Python SDK](https://docs.gretel.ai/guides/environment-setup/cli-and-sdk) and run `gretel configure` to configure API authentication.
+
+## Rate Limiting
+
+Stream creation (LLM requests) are rate limited to approximately 3 requests every 15 seconds. On average, you may create a stream every 5 seconds. If this limit is exceeded, the stream creation endpoint will return HTTP 403 errors.
+
+# API Usage
+
+These sections details the specifics of making API calls for Tabular LLM. All requests are made with `HTTP POST` requests.
```
Comment (matthewgrossman) [inline docs/inference_api_beta/README.md:29]:
```suggestion
These sections detail the specifics of making API calls for Tabular LLM. All requests are made with `HTTP POST` requests.
```

File: docs/inference_api_beta/README.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+## Make a LLM Request
+
+To submit a prompt to Tabular LLM, make the following call. This example uses one of the sample prompts from the Model Playground. Please update the prompt to suit your needs.
+
+```
+POST https://api.gretel.cloud/v1/inference/tabular/stream
+
+{
+  "model_id": "gretelai/tabular-v0",
```
Comment (matthewgrossman) [inline docs/inference_api_beta/README.md:39]: is it worth calling out the `/models` endpoint? or do we just want to direct people to `tabular-v0` anyways, and unless you are a partner, you have no idea what the difference between a/b/c is

File: docs/inference_api_beta/README.md (hunk 1 above)
Comment (matthewgrossman) [reply docs/inference_api_beta/README.md:13]: its still ending a sentence w/ a preposition so its not a much better fix, feel free to rephrase

**Review by matthewgrossman — APPROVED** [review]

File: docs/inference_api_beta/python/gretel.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+        print("Creating Gretel TabLLM inference stream.")
+        resp = _do_api_call("post", "/v1/inference/tabular/stream", body=payload)
+        self._curr_stream_id = resp.get("stream_id")
+        self._next_iter = None
+        print(f"Created Gretel TabLLM inference stream: {self._curr_stream_id}")
+
+    def iter_stream(self) -> Iterator[dict[str, Any]]:
+        while self.generated_count < self.target_count:
+            # Make an API request for a new Gretel stream if need be
```
Comment (stefan-kickoff) [inline docs/inference_api_beta/python/gretel.py:87]: How are we preventing duplicates after 50 rows if we're starting a new stream?

File: docs/inference_api_beta/README.md (hunk 3 above)
Comment (johntmyers) [reply docs/inference_api_beta/README.md:39]: Yeah I left that part out for now. We can add it if we need too. There's a helper function that pulls the model IDs, too.

File: docs/inference_api_beta/python/gretel.py (hunk 4 above)
Comment (johntmyers) [reply docs/inference_api_beta/python/gretel.py:87]: We don't

**Review by stefan-kickoff — APPROVED** [review]

**Pull request merged**